src/wamr/WAMRWasmModule.cpp

```diff
@@ -187,29 +199,139 @@ int WAMRWasmModule::executeWasmFunction(const std::string& funcName)
     } else {
         std::string errorMessage(
           ((AOTModuleInstance*)moduleInstance)->cur_exception);
-        SPDLOG_ERROR("Function failed: {}", errorMessage);
+        SPDLOG_ERROR("Caught wasm runtime exception: {}", errorMessage);
+
+        // TODO - agree on a protocol to send the return code through the error
+        // message (throwing exceptions not supported). For the moment, if the
+        // error message is a "Exception: 0" we assume no errors (i.e. exit(0))
```

Comment:
Is the `Exception: 0` string defined somewhere in WAMR or is that something we've created? Can we link to the WAMR constant if so? If not, we should create our own constant which should live in `#define WAMR_AOT_FAIL_MESSAGE "Exception: 0"` or something like that.

What is the scenario where the result of `aot_create_exec_env_and_call_function` is `false`, yet we're returning `0`, implying that the function was a success? Surely whenever `aot_create_exec_env_and_call_function` returns `false` we should at least return `1` (if not the actual error code).

If we control the format of this error message, it makes most sense to me just to have a string of the return code, then call `stoi` on it.

Reply:
There was a missunderstanding on my side, in which I mimicked the behaviour of `libc`'s `exit` rather than `wasi_proc_exit`.

Turns out that WAMR does the same I implemented but at the runtime level. I.e. WASI sets an exception with string "wasi proc exit", and, upon exit with exception, if the string is "wasi proc exit" WAMR removes said exception and returns 0.

See source here: https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/aot/aot_runtime.c#L874